opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -622,23 +630,30 @@ def start(
 
     def end(self, end_time: Optional[int] = None) -> None:
         with self._lock:
-            if not self.is_recording():
-                return
-            if self.start_time is None:
-                raise RuntimeError("Calling end() on a not started span.")
+            is_recording = self.is_recording()
```

Comment:
`is_recording()` is a public API. I believe methods within trace should work with `self._is_recording` so we don't depend on the implementation is `is_recording()`

Reply:
Fair point - will switch up all the internal calls to hit the attrib. I was following the prev pattern which internally used the method, but that was an artifact of it not having an internal attrib representation backing it.